### PR TITLE
Add/Update Location Forms (CU-m72ted)

### PR DESCRIPTION
**Pull request by risatronic**
- Added add and update forms and routes for location
- Removed/renamed db columns (from [this task](https://app.clickup.com/t/gez8x0))

I still need to write some tests but wanted to touch base with folks tomorrow on a few things first, mainly as regards what's actually mandatory and where checks should be done

### CODE REVIEW

**Comment by trisapeace** [comment]
Task linked: [CU-m72ted Create GUI for adding/updating locations](https://app.clickup.com/t/m72ted)

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,15 @@ the code was deployed.
 ## [Unreleased]
```
[the remark this answers is not in the document]
Comment (risatronic) [reply CHANGELOG.md:12]: Updated!

File: index.js [hunk 2]
```diff
@@ -452,6 +472,127 @@ app.get('/dashboard/:locationId', sessionChecker, async (req, res) => {
...
+      helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
+      res.status(400).send()
+    }
+  } catch (err) {
+    helpers.log(err)
+    res.status(500).send()
+  }
+})
+
+app.post('/locations/:locationId', Validator.body(['displayName', 'phone']).exists(), async (req, res) => {
```
[the remark this answers is not in the document]
Comment (risatronic) [reply index.js]: So this actually wound up being a massive pain for testing and I couldn't find an especially nice and easy workaround but none of the other post requests have it either so I feel like updating them/their tests is probably a separate task :/

File: index.js [hunk 3]
```diff
@@ -452,6 +472,127 @@ app.get('/dashboard/:locationId', sessionChecker, async (req, res) => {
   }
 })
 
+app.get('/locations/:locationId/edit', sessionChecker, async (req, res) => {
+  try {
+    const allLocations = await db.getLocations()
+    const currentLocation = await db.getLocationData(req.params.locationId)
```
[the remark this answers is not in the document]
Comment (risatronic) [reply index.js]: done :)

File: index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (trisapeace) [reply index.js]: We need this before we can merge or else anyone will be able to add/modify locations in our DB.

**Comment by trisapeace** [comment]
@mariocimet , I've made some changes to this and it is currently deployed on https://dev.sensors.brave.coop/ .

**Review by mariocimet — APPROVED** [review]

File: db/db.js [hunk 4]
```diff
@@ -540,62 +529,137 @@ async function getLocations() {
 }
 
 // Updates the locations table entry for a specific location with the new data
-// prettier-ignore
-async function updateLocationData(deviceid, phonenumber, detection_min, detection_max, sensitivity, noisemap, led, rpm_threshold, still_threshold, duration_threshold, mov_threshold, door_particlecoreid, radar_particlecoreid, radar_type, location) {
+// eslint-disable-next-line prettier/prettier
+async function updateLocation(
```
Comment (trisapeace) [inline db/db.js:533]: Note to self change this formatting back.

Comment (trisapeace) [reply db/db.js]: Fixed.

**Pull request merged**